### DISCUSSION TITLE
$preserveKeys does not work with ResourceClass::collection() when the collection keys are numeric and JsonResource::withoutWrapping().

### DISCUSSION THREAD

**Discussion by pantone170145** [laravel/framework#33731] [General] [answered]
- Laravel Version: 7.22.4
- PHP Version: 7.4.5
- Database Driver & Version: sqlite

### Description:

According to the documentation (https://laravel.com/docs/7.x/eloquent-resources#resource-collections , chapter 'Preserving Collection Keys'), to preserve keys in resource collection, you just need to set

```php
public $preserveKeys = true;
```

And then documentation (https://laravel.com/docs/7.x/eloquent-resources#data-wrapping) ,
I wanted to disable wrapping of the outermost resource, I made the following settings.

@AppServiceProvider.php
```php
public function boot()
{
    JsonResource::withoutWrapping();
}
```

As a result,
it works only for string keys, not for numeric keys

I didn't know if this behavior was correct

#### Collect case for string keys
input array
```
["str1" => "id":"1"]

```

output

```
{["str1" : "id":"1"]}

```

#### Incollect case for numeric keys
input array
```
["1" => "id":"1"]

```

output
```
{["id":"1"]}

```

**Comment by staudenmeir** [comment, marked as answer]
Please see #30052.

**Reply by pantone170145** [reply]
@staudenmeir Thank you for your information!
